scripts/upgrade/upgrade-facets.ts

```diff
@@ -0,0 +1,568 @@
+    const config: FacetConfig | undefined = fs.existsSync(configPath)
+      ? JSON.parse(fs.readFileSync(configPath, "utf8"))
+      : undefined;
+
+    if (config) {
+      console.log(`📋 Upgrade description: ${config.description}`);
+    } else {
+      console.log(`ℹ️  No upgrade config found for version ${version}, will only execute hooks if available`);
+    }
+
+    await executePreUpgradeHook(version, protocolAddress, originalChainId, originalEnv, dryRun);
+
+    if (config) {
+      await executeFacetUpdates(config, protocolAddress, contracts, contractsFile, isForkTest);
+    }
+
+    await executePostUpgradeHook(version, protocolAddress);
+
+    const initializationFacet = await ethers.getContractAt("InitializationFacet", protocolAddress);
+    const newVersion = (await initializationFacet.getVersion()).replace(/\0/g, "");
+    console.log(`\n📋 New protocol version: ${newVersion}`);
+
+    const contractsPath = await writeContracts(contracts, env, newVersion);
+    console.log(`✅ Contracts written to ${contractsPath}`);
+
+    if (dryRun) {
+      const balanceAfter = await getBalance();
+      const etherSpent = balanceBefore - balanceAfter;
+      console.log(`\n💰 Gas spent: ${ethers.formatEther(etherSpent)} ETH`);
+    }
+
+    console.log("\n🎉 Upgrade completed successfully!");
+  } catch (error) {
+    console.error("\n❌ Upgrade failed:", error);
+    throw error;
+  }
+}
+
+// Internal functions below
+async function getUserResponse(question: string, validResponses: string[]): Promise<string> {
+  console.error(question);
+  const answer = await new Promise<string>((resolve) => {
+    rl.question("", resolve);
+  });
+  if (validResponses.includes(answer)) {
+    return answer;
+  } else {
+    console.error("Invalid response!");
+    return getUserResponse(question, validResponses);
+  }
+}
+
+async function deployAndPrepareNewFacet(
+  facetName: string,
+  config: FacetConfig,
+  contracts: any[],
+  contractsFile: any,
+): Promise<DeployedFacet | null> {
+  console.log(`\n📦 Deploying ${facetName}...`);
+  const Facet = await hre.ethers.getContractFactory(facetName);
+  const functionSelectors = getSelectors(Facet);
+
+  if (functionSelectors.length === 0) {
+    console.log(`⚠️  Skipping ${facetName} as it has no function selectors`);
+    return null;
+  }
+
+  const skipSelectors = config.facets.skipSelectors?.[facetName] ?? [];
+  const filteredSelectors = removeSelectors(functionSelectors, skipSelectors);
+
+  if (filteredSelectors.length === 0) {
+    console.log(`⚠️  Skipping ${facetName} as all its selectors are in skip list`);
+    return null;
+  }
+
+  const constructorArgs = await prepareConstructorArgs(facetName, config, contracts, contractsFile);
+  const facetContract = await Facet.deploy(...constructorArgs);
+  await facetContract.waitForDeployment();
+  const facetAddress = await facetContract.getAddress();
+  console.log(`✅ ${facetName} deployed at: ${facetAddress}`);
```

Comment:
I couldn't find a place, where `contracts` is updated with the new address and constructor argument. This seems like a good place to do it.

And if you improve the `deploymentComplete` as I suggested in the other comment ([this thread](https://github.com/fermionprotocol/contracts/pull/451/files/8dc7e12811b6fdbaad949fe3704c697de700e83a#r2034709390)), it can be reused here, too.